TSG/Update/manual-retry-after-caupostcheck.md

```diff
@@ -0,0 +1,74 @@
+# Overview
+Since previous run failed at CauPostCheck step, ECE will always resume the action plan from previous failed step, it will not trigger a new update run. 
+
+To confirm if this article applies to your case, check the error below
+
+> "ErrorMessage": CloudEngine.Actions.InterfaceInvocationFailedException: Type 'EvalCauRetryApplicability' of Role 'CAU' raised an exception:
+> 
+> Year, Month, and Day parameters describe an un-representable DateTime.
+> at <ScriptBlock>, <No file>: line 8
+>    at CloudEngine.Actions.PowerShellHost.WaitAndReceiveJob(Job job, CancellationToken token, UInt32 timeoutSeconds, Stopwatch watch) in C:\__w\1\s\src\EceEngine\ece\CloudEngine\Actions\Source\PowerShellHost.cs:line 231
+>    at CloudEngine.Actions.PowerShellHost.Invoke(InterfaceParameters parameters, CancellationToken token, UInt32 timeoutSeconds, ThrottlingDescription throttlingDesc) in C:\__w\1\s\src\EceEngine\ece\CloudEngine\Actions\Source\PowerShellHost.cs:line 156
+>    at CloudEngine.Actions.InterfaceTask.Invoke(Configuration roleConfiguration, String startStep, String endStep, String[] skip, Nullable\`1 retries, Nullable\`1 interfaceTimeout, CancellationToken token, Dictionary`2 runtimeParameter, Boolean runInProcess) in C:\__w\1\s\src\EceEngine\ece\CloudEngine\Actions\Source\InterfaceTask.cs:line 236
+
+To mitigate this issue, we have to collect the LastUpdate xml and reset the CAU step. Then use 'invoke-actionplaninstance' cmd to trigger a new custom action plan. Please follow the instruction below to trigger a new custom action plan:
+
+# How to manually retry after failed at 'CauPostCheck' step
+1. Resolve the leftover status of previous failed run
+
+   `Invoke-CauRun -ForceRecovery -Force`
+
+2. Get the latest MAS Update xml file:
+   
+   `$ececlient = Create-ECEClusterServiceClient`
+
+   `$LastUpdate = Get-ActionPlanInstances -ececlient $ececlient | Where-Object ActionPlanName -match "MAS Update" | Sort-Object -Property EndDateTime -Descending | Select-Object -First 1`
+
+   `$LastUpdate.ProgressAsXml | Out-File c:\temp\LastUpdate.xml`
+
+   Once you get the LastUpdate xml file, if the status of step '20.20.8.20.20.4' is not 'Error', you might collect the wrong xml file. You could use below command listing all action instance and find the one which 'actionplanname' is 'MAS Update' and the time is matched with last failure. The status of the action plan should be 'Failed':
+
+   `Get-actionplaninstances -ececlient $ececlient | ft status, startdatetime, enddatetime, instanceid, actionplanname`
+
+   Then use that 'actionplaninstancid' to save the correct LastUpdate xml file again by using this command:
+
+   `$(get-actionplaninstance -actionplaninstanceid <instanceid> -ececlient $ececlient).ProgressAsXml | Out-File -FilePath c:\temp\LastUpdate.xml`
+
+3. Open the LastUpdate.xml file in c:\temp folder and find the step '20.20.8.20.20.4'. Replace the whole content of step '20.20.8.20.20.4' by this content:
```

Comment:
I'm a bit hesitant to open up this relatively complicated process to customers to attempt without involvement from CSS. There are many touch points and corner cases here that could lead to a messy incident to untangle if the customer does something wrong.

Reply:
@alexbass112 do you want me to completely exclude this document?